buildbot_nix/__init__.py

```diff
@@ -756,16 +756,57 @@ def __init__(self, project: GitProject, path: Path, **kwargs: Any) -> None:
         self.project = project
         self.path = path
 
+    def join_traversalsafe(self, root: Path, joined: Path) -> Path:
+        root = root.resolve()
+
+        for part in joined.parts:
+            new_root = (root / part).resolve()
+
+            if not new_root.is_relative_to(root):
+                msg = f"Joined path attempted to traverse upwards when processing {root} against {part} (gave {new_root})"
+                raise ValueError(msg)
+
+            root = new_root
+
+        return root
+
+    def join_all_traversalsafe(self, root: Path, *paths: Path) -> Path:
+        for path in paths:
+            root = self.join_traversalsafe(root, path)
+
+        return root
+
+    @defer.inlineCallbacks
     def run(self) -> Generator[Any, object, Any]:
         props = self.build.getProperties()
-        if props.getProperty("branch") != self.project.default_branch:
+
+        pr = props.getProperty("pr_number")
```

Comment:
Ah I forgot, why I couldn't merge this. We don't have this in github.

Reply:
Would be useful to have this for outputs on the default branch, can PRs be dropped? e.g.

```diff
diff --git a/buildbot_nix/__init__.py b/buildbot_nix/__init__.py
index 3dd09f3..1d6dec5 100644
--- a/buildbot_nix/__init__.py
+++ b/buildbot_nix/__init__.py
@@ -780,9 +780,7 @@ class UpdateBuildOutput(steps.BuildStep):
     def run(self) -> Generator[Any, object, Any]:
         props = self.build.getProperties()
 
-        pr = props.getProperty("pr_number")
-
-        if not pr and props.getProperty("branch") != self.project.default_branch:
+        if props.getProperty("branch") != self.project.default_branch:
             return util.SKIPPED
 
         out_path = props.getProperty("out_path")
@@ -792,7 +790,7 @@ class UpdateBuildOutput(steps.BuildStep):
 
         project_name = Path(props.getProperty("projectname"))
 
-        target = Path(f"pulls/{pr}") if pr else Path(props.getProperty("branch"))
+        target = Path(props.getProperty("branch"))
 
         attr = Path(props.getProperty("attr"))
 

```

Reply:
yes. Sounds like a good idea.